overlord/snapstate/handlers.go

```diff
@@ -166,8 +169,25 @@ func (m *SnapManager) doPrerequisites(t *state.Task, _ *tomb.Tomb) error {
 		prereqName = snapsup.Base
 	}
 
+	// Ensure prereq from base snaps are available here.
+	if err := m.installPrereq(t, prereqName, snapsup.UserID); err != nil {
+		return err
+	}
+	// Ensure other prereqs are available here, i.e. content snaps
+	for _, prereqName := range snapsup.Prereq {
+		if err := m.installPrereq(t, prereqName, snapsup.UserID); err != nil {
+			return err
+		}
+	}
```

Comment:
I'm a bit worried that the Retry returned from installPrereq and this sort of looping is not compatible. If we keep the retry approach we should not add tasks to the change if we are not sure we are about to be able to be done and get out of the prerequisite task step. Otherwise we might get cross waits that will never succeed if we are not careful. These are similar problems to trying to get multiple locks at the same time, that is always delicate.

Reply:
I can make an example tomorrow if needed